core/src/main/java/org/apache/calcite/rel/rules/JoinCommuteRule.java

```diff
@@ -137,6 +137,12 @@ public JoinCommuteRule(Class<? extends Join> clazz,
         .build();
   }
 
+  public static RexNode swapJoinCond(RexNode cond, Join join, RexBuilder rexBuilder) {
```

Comment:
Does this have to be public or can it be moved to the file where it's used?

Reply:
It relies on VariableReplacer, which is a private class, so I wrap a method here.